tesseract/tesseract_common/include/tesseract_common/resource.h

```diff
@@ -0,0 +1,31 @@
+#ifndef TESSERACT_COMMON_RESOURCE_H
+#define TESSERACT_COMMON_RESOURCE_H
+
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <vector>
+#include <memory>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+namespace tesseract_common
+{
+// Represents resource data available from a file or url
+class Resource
+{
+public:
+  using Ptr = std::shared_ptr<Resource>;
+  using ConstPtr = std::shared_ptr<const Resource>;
+
+  virtual bool IsFile() = 0;
```

Comment:
I recommend renaming the functions to match the ROS Cpp style guide. This function would be `isFile()`.

Reply:
Also should all these functions be const?

Reply:
I will update to use the ROS Cpp style.

No, I am not using const because it is likely that the located resources will be modifying cache data during operations.

Reply:
The style should be updated now